pkg/component/autoscaling/clusterautoscaler/cluster_autoscaler.go

```diff
@@ -617,11 +637,58 @@ func (c *clusterAutoscaler) computeShootResourcesData(serviceAccountName string)
 			},
 		}
 	)
+	objects := []client.Object{clusterRole, clusterRoleBinding, role, rolebinding}
 
-	return registry.AddAllAndSerialize(
-		clusterRole,
-		clusterRoleBinding,
-		role,
-		rolebinding,
-	)
+	if workersHavePrioritiesConfigured {
+		configMap, err := c.generatePriorityExpanderConfigMap()
+		if err != nil {
+			return nil, err
+		}
+		objects = append(objects, configMap)
+	}
+	return registry.AddAllAndSerialize(objects...)
+}
+
+func (c *clusterAutoscaler) generatePriorityExpanderConfigMap() (*corev1.ConfigMap, error) {
+	priorities := map[int32][]string{}
+	for _, machineDeployment := range c.machineDeployments {
+		// TODO(tobschli): Remove this once all well-known extensions have revendored to use the generic actuator that sets the priorities.
+		if machineDeployment.Priority == nil {
+			// In this case, the extension did not set the priorities that were configured in the worker.
+			// fall back to try to determine the pool name.
+			for _, pool := range c.workerConfig {
```

Comment:
This double loop could be slow for clusters with large number of nodes. What about using a `map[string]int32` from `pool.Name` to `ptr.Deref(pool.Priority, 0)`, then getting the name of the pool from the `machineDeployment.Name` like this:
```
name := strings.TrimPrefix(machineDeployment.Name, c.namespace+"-")
zoneIndex := strings.LastIndex(name, "-z")
if zoneIndex != -1 {
    name = name[:zoneIndex]
}
``` 

and finally retrieving the priority from the map?